### ISSUE TITLE
Microsoft.CodeAnalysis.FxCopAnalyzers causes slow build

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn-analyzers#3533]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/911676/microsoftcodeanalysisfxcopanalyzers-causes-slow-bu.html)._

---
[regression]
a small c# library with Microsoft.CodeAnalysis.FxCopAnalyzers enabled builds very very slow. It takes about 5-10 minutes to compile. After disabling the Code Analyaer it takes sub seconds to compile. There were over 300 warnings with FXCop. 

This only occured in VS2019 16.4.4. The previous version did not have this issue.

---
#### Visual Studio Feedback System on 2/9/2020, 07:23 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Visual Studio Feedback System on 2/11/2020, 04:58 PM: 

<p>Thank you for sharing your feedback! Our teams prioritize action on product issues with broad customer impact. See details at: <a target='_blank' href="https://docs.microsoft.com/en-us/visualstudio/ide/report-a-problem?view=vs-2017#faq">https://docs.microsoft.com/en-us/visualstudio/ide/report-a-problem?view=vs-2017#faq</a>. In case you need answers to common questions or need assisted support, be sure to use <a target='_blank' href="https://visualstudio.microsoft.com/vs/support/">https://visualstudio.microsoft.com/vs/support/</a>. We’ll keep you posted on any updates to this feedback.</p>

#### Visual Studio Feedback System on 3/19/2020, 07:30 PM: 

I have detected that for the last 35 days, this issue didn't have much product team activity and a very small amount of new votes or comments. Based on this, its severity, and affected area, it’s my experience that this issue is very unlikely to be fixed.

---
### Original Solutions
(no solutions)

**Comment by Micktommyord** [comment]
can we please have some sort of fix for this please. The issues is definitely prominent when FxCopAnalyzers is installed on a EF Core project. especially when you have Migration folders for EF Migrations. it can take my pc up to 5 mins for a build. but without FXCopAnalyzers it is as quick as 20-25 seconds

**Comment by sharwell** [comment]
@mavasani I wasn't able to find a documentation page for the `reportAnalyzer` property. Can you add a link to instructions for identifying the slow analyzer(s)?

**Comment by mavasani** [comment]
https://github.com/dotnet/roslyn/blob/master/docs/analyzers/Report%20Analyzer%20Format.md

**Comment by mavasani** [comment]
This will likely get addressed with https://github.com/dotnet/roslyn-analyzers/pull/3788. I will point you to a package to try out once that PR is merged. Thanks!

**Comment by mavasani** [comment]
Can you please try out https://dev.azure.com/dnceng/public/_packaging?_a=package&feed=dotnet5&package=Microsoft.CodeAnalysis.FxCopAnalyzers&protocolType=NuGet&version=3.3.0-beta1.20326.1&view=overview?

Also, if that doesn't help, can you check if adding `dotnet_code_quality.interprocedural_analysis_kind = None` to your .editorconfig resolves the issue? Thanks!

**Comment by negberts** [comment]
We have the same problem, updating to the latest beta and adding the line to the .editorconfig file did not help. It appears that the build get's stuck because it tries to get the same resources while building multiple projects at once.  What was the setting to prevent projects from building simultaneously? So I want them to build in sequential order?

**Comment by negberts** [comment]
It typically hangs here:

Using shared compilation with compiler from directory: C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\Roslyn

**Comment by StarKiller54** [comment]
Hello, i have the same problem and `dotnet_code_quality.interprocedural_analysis_kind = None` work for me.
I have asp.net project with framework 4.5.2, visual studio 2019 16.7.1 and nuget package fxcop analyser 3.3.0.

**Comment by Evangelink** [comment]
@mavasani I think this ticket can be closed. If there is still some issue people will probable get back and open new ticket. WDYT?

**Comment by sharwell** [comment]
It's not possible to review general performance issues like this through GitHub. For users encountering a similar problem, there are two ways to help us get it resolved:

* For simple reproducible cases (e.g. "here is a specific source file which is slow with analyzer _name_ enabled"), a new issue with detailed steps to reproduce can be filed
* For other cases, the Report a Problem tool should be used to file a performance feedback using these steps: https://docs.microsoft.com/en-us/visualstudio/ide/how-to-increase-chances-of-performance-issue-being-fixed?view=vs-2019#slowness-and-high-cpu-issues
